test/framework/src/main/java/org/elasticsearch/action/support/single/shard/SingleShardRequestHelper.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.support.single.shard;
+
+import org.elasticsearch.index.shard.ShardId;
+
+public class SingleShardRequestHelper {
+
+    private SingleShardRequestHelper() {}
+
+    public static void setInternalShardId(SingleShardRequest<?> singleShardRequest, ShardId shardId) {
```

Comment:
Seems a bit off to introduce production code to allow invalid states only used in tests. I have an alternative idea will ping you about.